webapp/src/components/ProgramActivityDialog.tsx

```diff
@@ -1,85 +1,212 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { decodeHTML } from 'entities';
+import CancelIcon from '@mui/icons-material/Cancel';
+import CloseIcon from '@mui/icons-material/Close';
+import TaskAltIcon from '@mui/icons-material/TaskAlt';
+import LoadingButton from '@mui/lab/LoadingButton';
 import {
-  Button,
+  Alert,
   Dialog,
   DialogActions,
   DialogContent,
   DialogContentText,
   DialogTitle,
   Divider,
+  FormGroup,
+  Snackbar,
   Table,
   TableBody,
   TableCell,
   TableContainer,
   TableRow,
 } from '@mui/material';
-
+import IconButton from '@mui/material/IconButton';
 import { formatDate, formatTime } from '../helpers/dateTime';
 import { CalendarEvent } from '../types/api';
-
 interface ProgramActivityDialogProps {
   show: boolean;
   contents: CalendarEvent;
   handleClose: () => void;
 }
+const sendAPIGetRequest = (
+  path: string,
+  activityType: string,
+  setCompleted: React.Dispatch<React.SetStateAction<boolean | null>>,
+  setError: React.Dispatch<React.SetStateAction<boolean | null>>,
+  setIsErrorShown: React.Dispatch<React.SetStateAction<boolean>>,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+) => {
+  setIsLoading(true);
+  if (activityType !== 'assignment') return;
+  return fetch(`${process.env.REACT_APP_API_ORIGIN}${path}`, {
+    method: 'GET',
+    credentials: 'include',
+  })
+    .then(res => res.json())
+    .then(
+      async ({ data }) => {
```

Comment:
Does this need to be an async function? It isn't for the PUT request version of this function.